### PR TITLE
added a function to the symbol trait called "native_symbol" 

### LINKED ISSUE #493: CoffSymbol cant access underlying ImageSymbol field
Hi id like to get access to the storage_type and derived_id fields of the ImageSymbol structure when it is encapsulated into CoffSymbol... Seems you only have wrapper methods that return some higher level information. 

Maybe I am missing something, can i convert a CoffSymbol into a ImageSymbol?

Thanks.

References:
https://docs.rs/object/latest/object/read/coff/struct.CoffSymbol.html (CoffSymbol)
https://docs.rs/object/0.30.0/src/object/read/coff/symbol.rs.html#325 (id like access to this field)
https://docs.rs/object/0.30.0/src/object/pe.rs.html#791 (and this field)

Possible Solution:
https://docs.rs/object/latest/object/read/trait.ObjectSymbol.html#provided-methods

Add another provided method that returns the underlying symbol (clone it)...

`fn native_symbol(&self) -> NativeSymbolType` where you can do `type NativeSymbolType` and whomever impliments ObjectSymbol needs to set the NativeSymbolType (for example ImageSymbol for coff).

### PR DESCRIPTION
**Pull request by invlpg**
This allows you to get a copy of the underlying native symbol structure. 

Closes #493

### CODE REVIEW

**Comment by invlpg** [comment]
https://github.com/gimli-rs/object/issues/493

**Comment by invlpg** [comment]
Ready to merge

File: src/read/coff/symbol.rs [hunk 1]
```diff
@@ -325,6 +325,14 @@ where
     pub(crate) symbol: &'data pe::ImageSymbol,
 }
 
+impl<'data, 'file, R: ReadRef<'data>> CoffSymbol<'data, 'file, R> {
+    #[inline]
+    /// get the raw ImageSymbol struct.
```
Comment (philipc) [inline src/read/coff/symbol.rs:330]:
```suggestion
    /// Get the raw `ImageSymbol` struct.
```

File: src/read/coff/symbol.rs [hunk 2]
```diff
@@ -325,6 +325,14 @@ where
...
+    pub fn raw_symbol(&self) -> pe::ImageSymbol {
+        *self.symbol
```
Comment (philipc) [inline src/read/coff/symbol.rs:332]:
```suggestion
    pub fn raw_symbol(&self) -> &'data pe::ImageSymbol {
        self.symbol
```

File: tests/read/coff.rs [hunk 3]
```diff
@@ -21,3 +21,20 @@ fn coff_extended_relocations() {
     let relocations = code_section.relocations().collect::<Vec<_>>();
     assert_eq!(relocations.len(), 65536);
 }
+
+
+#[cfg(feature = "coff")]
+#[test]
+fn coff_symbol_raw_test() {
```
Comment (philipc) [inline tests/read/coff.rs:28]: Thanks for the diligence with regard to testing. However, I prefer not to have tests for trivial getter functions, so let's leave this out.

**Review by philipc — CHANGES_REQUESTED** [review]

**Comment by invlpg** [comment]
Ill make these changes real quick

**Review by invlpg — COMMENTED** [review]
cool

File: tests/read/coff.rs [hunk 4]
```diff
@@ -20,4 +20,4 @@ fn coff_extended_relocations() {
     };
     let relocations = code_section.relocations().collect::<Vec<_>>();
     assert_eq!(relocations.len(), 65536);
-}
+}
```
Comment (philipc) [inline tests/read/coff.rs:23]: You need to revert this change.

**Comment by invlpg** [comment]
ok ran cargo fmt

**Review by philipc — APPROVED** [review]

**Comment by philipc** [comment]
Coverage failure is probably https://github.com/actions-rs/tarpaulin/issues/21

**Pull request merged**